File: zhtml/src/com/potix/zhtml/Input.java
/* Input.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Tue Nov 29 21:59:11     2005, Created by tomyeh@potix.com
}}IS_NOTE

Copyright (C) 2005 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 2.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
*/
package com.potix.zhtml;

import com.potix.zk.ui.WrongValueException;
import com.potix.zk.ui.ext.Inputable;
import com.potix.zhtml.impl.AbstractTag;

/**
 * The input tag.
 *
 * @author <a href="mailto:tomyeh@potix.com">tomyeh@potix.com</a>
 */
public class Input extends AbstractTag implements Inputable {
	public Input() {
		super("input");
	}
	protected Input(String tagnm) {
		super(tagnm);
	}

	/** Returns the value of this input.
	 */
	public String getValue() {
		return (String)getDynamicProperty("value");
	}
	/** Sets the vallue of this input.
	 */
	public void setValue(String value) throws WrongValueException {
		setDynamicProperty("value", value);
	}

	//-- Inputable --//
	public void setTextByClient(String value) throws WrongValueException {
		setValue(value);
	}
}

File: zul/src/archive/web/js/zul/inp/mold/input.js
/* input.js

	Purpose:
		
	Description:
		
	History:
		Fri Jan 16 13:13:15     2009, Created by tomyeh

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 3.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
function (out) {
	var zcls = this.getZclass(),
		uuid = this.uuid,
		isRounded = this.inRoundedMold();
	if(!isRounded)
		out.push('<input', this.domAttrs_(), '/>');
	else {
		out.push('<i', this.domAttrs_({text:true}), '>',
				'<input id="', uuid, '-real"', 'class="', zcls, '-inp"', 
				this.textAttrs_(), '/>', '<i id="', uuid, '-right-edge"',
				'class="', zcls, '-right-edge');
		if (zk.ie6_ && this._readonly)
			out.push(' ', zcls, '-right-edge-readonly');
		out.push('"></i></i>');
	}
}